zipline/assets/assets.py

```diff
@@ -624,10 +681,16 @@ def _retrieve_asset_dicts(self, sids, asset_tbl, querying_equities):
 
         if querying_equities:
             def mkdict(row,
+                       exchanges=self.exchange_info,
                        symbols=self._lookup_most_recent_symbols(sids)):
-                return merge(row, symbols[row['sid']])
+                d = dict(row)
```

Comment:
we do call this function on every single symbol->sid record. This code is already super slow, but not because we have one single hot function, we are just bleeding small losses all over. I could make it a non-local but idk if it makes it any less readable this way.